packages/container/libraries/core/src/metadata/calculations/getClassElementMetadataFromLegacyMetadata.ts

```diff
@@ -0,0 +1,82 @@
+import { ServiceIdentifier } from '@inversifyjs/common';
+
+import {
+  INJECT_TAG,
+  MULTI_INJECT_TAG,
+  NAME_TAG,
+  NAMED_TAG,
+  NON_CUSTOM_TAG_KEYS,
+  OPTIONAL_TAG,
+  UNMANAGED_TAG,
+} from '../../reflectMetadata/data/keys';
+import { ClassElementMetadata } from '../models/ClassElementMetadata';
+import { ClassElementMetadataKind } from '../models/ClassElementMetadataKind';
+import { LegacyMetadata } from '../models/LegacyMetadata';
+import { ManagedClassElementMetadata } from '../models/ManagedClassElementMetadata';
+import { MetadataName } from '../models/MetadataName';
+import { MetadataTag } from '../models/MetadataTag';
+import { MetadataTargetName } from '../models/MetadataTargetName';
+
+export function getClassElementMetadataFromLegacyMetadata(
+  metadataList: LegacyMetadata[],
+): ClassElementMetadata {
+  const injectMetadata: LegacyMetadata | undefined = metadataList.find(
+    (metadata: LegacyMetadata): boolean => metadata.key === INJECT_TAG,
+  );
+  const multiInjectMetadata: LegacyMetadata | undefined = metadataList.find(
+    (metadata: LegacyMetadata): boolean => metadata.key === MULTI_INJECT_TAG,
+  );
+  const unmanagedMetadata: LegacyMetadata | undefined = metadataList.find(
+    (metadata: LegacyMetadata): boolean => metadata.key === UNMANAGED_TAG,
+  );
+
+  if (unmanagedMetadata !== undefined) {
+    return {
+      kind: ClassElementMetadataKind.unmanaged,
+    };
+  }
+
+  if (multiInjectMetadata === undefined && injectMetadata === undefined) {
+    throw new Error();
```

Comment:
Consider this as sample code, we should throw a meaningful error indicating either `inject` `multiInject` or `unmanaged` must be included

Reply:
Hi @notaphplover 

I read through your proposal and think it makes a lot of sense—it would definitely be a valuable addition to the Inversify core. One point that could enhance your proposal is elaborating a bit more on the project structure. Specifically, how it would look if we were to implement additional packages, such as inversify-bun-utils, inversify-express-utils, or inversify-fastify-utils, with the core package containing the shared decorators for these HTTP frameworks.

Another aspect to consider is a modular approach with:

@inversify/core
@inversify/common

The common module could include utilities that are shared across implementations, such as decorators, without requiring the full core package. This would enable a lightweight, flexible option for those implementing plugins or working within different frameworks, as they could pull from common without the overhead of the entire core.

Overall, the proposal looks strong, and these additional considerations could make it even more versatile. Great work!